src/datasets/arrow_dataset.py

```diff
@@ -3358,6 +3358,468 @@ def init_buffer_and_writer():
+             desc (`str`, *optional*, defaults to `None`):
+                 Meaningful description to be displayed alongside with the progress bar while mapping examples.
+
+         Example:
+
+         ```py
+         >>> from datasets import load_dataset
+         >>> from collections import Counter
+
+         # Count the number of words in the validation set of the Rotten Tomatoes dataset
+         >>> rotten_ds = load_dataset("rotten_tomatoes", split="validation")
+         >>> count = lambda counter, text: counter + Counter(text.lower().split())
+         >>> sum_counts = lambda counter1, counter2: counter1 + counter2
+         >>> result = result = rotten_ds.select(range(1)).reduce(count, initializer=Counter(), input_columns="text")
+         >>> result
+        {'text': Counter({'and': 2, 'compassionately': 1, 'explores': 1, 'the': 1, 'seemingly': 1, 'irreconcilable': 1, 'situation': 1, 'between': 1, 'conservative': 1, 'christian': 1, 'parents': 1, 'their': 1, 'estranged': 1, 'gay': 1, 'lesbian': 1, 'children': 1, '.': 1})}
+        
+         # Calculate the average number of stars in the "Video_Games_v1_00" subset of the Amazon US reviews dataset
+         >>> review_ds = load_dataset("amazon_us_reviews", "Video_Games_v1_00", split="train")
+         >>> sum = lambda accumulator, review: accumulator + review
+         >>> sum_sums = lambda sum1, sum2: sum1 + sum2
+         >>> result = review_ds.reduce(sum, combiner=sum_sums, initializer=0, input_columns="star_rating")
+         >>> result['star_rating'] / length(review_ds)
+        4.059892597803915
+
+         # process a batch of examples
+         >>> review_ds = review_ds.reduce(sum, combiner=sum_sums, initializer=0, input_columns="star_rating", batched=True)
+
+         # set number of processors
+         >>> result = review_ds.reduce(sum, combiner=sum_sums, initializer=0, input_columns="star_rating", num_proc=4)
+
+         # set number of processors, with non-empty initializer, for input type `int`
+         >>> int_ds = Dataset.from_dict({"x": [1, 2, 3]})
+         >>> sum_reduce = lambda x, y: x + y
+         >>> reduction = int_ds.reduce(sum_reduce, combiner=sum_reduce, initializer=1, input_columns='x', num_proc=2)
+         >>> reduction       # reduction is (1+(1+2)+1+(3)) = 8, i.e. initializer + reduction_1 + initializer + reduction_2
+        {'x': 8}
+
+         # set number of processors, with non-empty initializer, for input type `str`
+         >>> string_ds = Dataset.from_dict({"x": ["1", "2", "3"]})
+         >>> sum_reduce_reverse = lambda x, y: y + x
+         >>> reduction = string_ds.reduce(sum_reduce, combiner=sum_reduce_reverse, initializer="X", input_columns='x', num_proc=2)
+         >>> reduction       # reduction is ('X'+('3')+'X'+('2'+'1')) = 'X3X12', i.e. `initializer + reduction_2 + initializer + reduction_1`, note that the order of the reduction is reversed
+         {'x': 'X3X12'}
```

Comment:
I think we can remove this one - we're explaining the same thing as in the previous example